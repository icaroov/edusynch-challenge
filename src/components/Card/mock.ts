export default [
  {
    id: '1',
    image:
      'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
    numberLessons: 10,
    description: 'Master English: Improve Your Speaking'
  },
  {
    id: '2',
    image:
      'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
    numberLessons: 3,
    description: 'Master English: Improve Your Speaking'
  },
  {
    id: '3',
    image:
      'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
    numberLessons: 1,
    description: 'Master English: Improve Your Speaking'
  },
  {
    id: '4',
    image:
      'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
    numberLessons: 4,
    description: 'Master English: Improve Your Speaking'
  },
  {
    id: '5',
    image:
      'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
    numberLessons: 3,
    description: 'Master English: Improve Your Speaking'
  },
  {
    id: '6',
    image:
      'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
    numberLessons: 6,
    description: 'Master English: Improve Your Speaking'
  }
]
