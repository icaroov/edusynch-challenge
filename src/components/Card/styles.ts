import styled, { css } from 'styled-components'

export const Container = styled.article`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;

    width: 100%;
    height: 100%;

    padding: 1rem;

    border-radius: ${theme.border.radius};
    background-color: ${theme.colors.white};
  `}
`

export const ImageBox = styled.div`
  ${({ theme }) => css`
    img {
      width: 100%;
      min-height: 165px;

      border-radius: ${theme.border.radius};
    }
  `}
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  height: 100%;
  margin-top: 1.4rem;
`

export const Info = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`

export const Stars = styled.div`
  text-decoration: none;
`

export const ButtonCard = styled.button`
  ${({ theme }) => css`
    padding: 1rem;
    color: ${theme.colors.black};

    font-size: ${theme.font.sizes.xsmall};
    background: ${theme.colors.lighPink};

    border: none;
    border-radius: ${theme.border.radius};

    cursor: help;
  `}
`

export const Description = styled.div`
  ${({ theme }) => css`
    margin: 1.6rem 0;

    font-family: ${theme.font.secondary};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};

    color: ${theme.colors.darkBg};
    text-transform: capitalize;
  `}
`
