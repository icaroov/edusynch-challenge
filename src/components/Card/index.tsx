import Rating from 'components/Rating'

import * as Styled from './styles'

export type CardProps = {
  description: string
  numberLessons: number
  image: string
}

const Card = ({ description, image, numberLessons }: CardProps) => (
  <Styled.Container>
    <Styled.ImageBox>
      <img src={image} alt={description} />
    </Styled.ImageBox>

    <Styled.Content>
      <Styled.Info>
        <Styled.Stars>
          <Rating />
        </Styled.Stars>
        <Styled.ButtonCard>{numberLessons} lessons</Styled.ButtonCard>
      </Styled.Info>

      <Styled.Description>
        <p>{description}</p>
      </Styled.Description>
    </Styled.Content>
  </Styled.Container>
)

export default Card
