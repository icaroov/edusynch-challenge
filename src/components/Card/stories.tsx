import { Story, Meta } from '@storybook/react/types-6-0'

import Card, { CardProps } from '.'

export default {
  title: 'Card',
  component: Card,
  args: {
    image:
      'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
    numberLessons: 10,
    description: 'Master English: Improve Your Speaking'
  }
} as Meta

export const Default: Story<CardProps> = args => <Card {...args} />

Default.parameters = {
  layout: 'fullscreen',
  backgrounds: {
    default: 'dark'
  }
}
