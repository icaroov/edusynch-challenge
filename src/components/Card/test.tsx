import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Card from '.'

const props = {
  image:
    'https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80',
  numberLessons: 10,
  description: 'Master English: Improve Your Speaking'
}

describe('<Card />', () => {
  it('should render the card', () => {
    const { container } = renderWithTheme(<Card {...props} />)

    expect(screen.getByRole('article')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
