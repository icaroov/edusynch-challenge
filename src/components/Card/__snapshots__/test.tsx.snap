// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the card 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  max-width: 360px;
  height: 100%;
  padding: 1rem;
  border-radius: 0.4rem;
  background-color: #FFFFFF;
}

.c1 img {
  width: 100%;
  min-height: 165px;
  border-radius: 0.4rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  margin-top: 1.4rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  padding: 1rem;
  color: #030517;
  font-size: 1.2rem;
  background: #CFC9E9;
  border: none;
  border-radius: 0.4rem;
  cursor: help;
}

.c7 {
  margin: 1.6rem 0;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.8rem;
  font-weight: 700;
  color: #200E64;
  text-transform: capitalize;
}

<article
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Master English: Improve Your Speaking"
      src="https://images.unsplash.com/photo-1513104890138-7c749659a591?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=750&q=80"
    />
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg
            color="#C4C4C4"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(196, 196, 196); margin-right: 5px; cursor: pointer;"
            viewBox="0 0 576 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            color="#C4C4C4"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(196, 196, 196); margin-right: 5px; cursor: pointer;"
            viewBox="0 0 576 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            color="#C4C4C4"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(196, 196, 196); margin-right: 5px; cursor: pointer;"
            viewBox="0 0 576 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            color="#C4C4C4"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(196, 196, 196); margin-right: 5px; cursor: pointer;"
            viewBox="0 0 576 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <svg
            color="#C4C4C4"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(196, 196, 196); margin-right: 5px; cursor: pointer;"
            viewBox="0 0 576 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
        </div>
      </div>
      <button
        class="c6"
      >
        10
         lessons
      </button>
    </div>
    <div
      class="c7"
    >
      <p>
        Master English: Improve Your Speaking
      </p>
    </div>
  </div>
</article>
`;
